### PR TITLE
Port Vegasettings to GTK3

### PR DESCRIPTION
**Pull request by DMJC**
I've updated vssetup to build using GTK3

### CODE REVIEW

File: engine/setup/CMakeLists.txt [hunk 1]
```diff
@@ -9,18 +9,18 @@ SET(VEGASETTINGS_SOURCES src/c/setup.cpp
                     ../src/common/common.cpp )                                                                                                        
 
 
-add_executable(vegasettings ${VEGASETTINGS_SOURCES})
+add_executable(vssetup ${VSSETUP_SOURCES})
```
Comment (BenjamenMeyer) [inline engine/setup/CMakeLists.txt:12]: This changed in master - both the executable name and the variable name.

File: engine/setup/CMakeLists.txt [hunk 2]
```diff
@@ -9,18 +9,18 @@ SET(VEGASETTINGS_SOURCES src/c/setup.cpp
...
-#find GTK2
-find_package(GTK2 REQUIRED)
-IF(GTK2_FOUND)
-    SET(GTK_LIBS ${GTK2_LIBRARIES})
-    SET(GTK_CFLAGS ${GTK2_DEFINITIONS} -DGTK)
-ELSE(GTK2_FOUND)
-     message("Cant get GTK2 found")
-ENDIF(GTK2_FOUND)
+#find GTK3
+find_package(GTK3 REQUIRED)
```
Comment (BenjamenMeyer) [inline engine/setup/CMakeLists.txt:15]: Is this supported by cmake 3.5?

**Comment by DMJC** [comment]
Updated to resolve some of the issues.

File: engine/setup/CMakeLists.txt (hunk 2 above)
Comment (DMJC) [reply engine/setup/CMakeLists.txt:15]: Not, sure I'm using Cmake 3.5

**Comment by nabaco** [comment]
Looks good, but:
- We need to update `.travis.yml` and `README` with the installation of the correct dependency. This is critical both for CI and in terms of release
- You haven't added `FindGTK3.cmake` to the commit (probably forgot to do `git add`)
- Could you please attach a picture of the final result, and list the tests you have done on this?

File: engine/setup/CMakeLists.txt (hunk 2 above)
Comment (BenjamenMeyer) [reply engine/setup/CMakeLists.txt:15]: based on the CI I'd say it's not, meaning we need something to define it.

File: engine/FindGTK2.cmake [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+#  GTK2_DEFINITIONS - Compiler switches required for using GTK2
+#
+#  Copyright (c) 2007 Andreas Schneider <mail@cynapses.org>
+#
+#  Redistribution and use is allowed according to the terms of the New
+#  BSD license.
+#  For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+#
+if (NOT BEOS)
+set(GTK2_DEBUG OFF)
```
Comment (BenjamenMeyer) [inline engine/FindGTK2.cmake]: This needs to be migrated to GTK3

File: engine/setup/CMakeLists.txt (hunk 2 above)
Comment (BenjamenMeyer) [reply engine/setup/CMakeLists.txt:15]: @dmjc thanks for including the related CMake file

**Comment by BenjamenMeyer** [comment]
I'm good once it passes CI.

**Comment by BenjamenMeyer** [comment]
@stephengtuggy can you check master to confirm if it's this PR or master?

If it's this PR then it need to be fixed here. If master we can merge, file a bug, and fix after merge.

**Comment by DMJC** [comment]
Can you post a screenshot of vegasettings running on your system?

**Comment by stephengtuggy** [comment]
> @stephengtuggy can you check master to confirm if it's this PR or master?
> 
> If it's this PR then it need to be fixed here. If master we can merge, file a bug, and fix after merge.

Done. It's this PR.

**Comment by stephengtuggy** [comment]
`master` is fine.

**Comment by stephengtuggy** [comment]
> Can you post a screenshot of vegasettings running on your system?

Before:
vegasettings before gtk3 PR

After:
vegasettings after gtk3 PR

Note that this is on Ubuntu 20.04 LTS.

**Comment by BenjamenMeyer** [comment]
Probably a difference in how the controls work, but certainly needs fixing before a merge

**Comment by nabaco** [comment]
@DMJC I recommend you to use [this version](https://raw.githubusercontent.com/Ettercap/ettercap/master/cmake/Modules/FindGTK3.cmake) of `FindGTK3.cmake` as it was tested on Windows too. I have use it in my work on #53, since your version didn't work for me.

**Comment by BenjamenMeyer** [comment]
@DMJC Just FYI - https://github.com/vegastrike/Vega-Strike-Engine-Source/pull/133, modifies the CMakeLists.txt for consistency in formatting.

**Comment by nabaco** [comment]
This is how I see it on my system, will try to fix:
W_2020-06-07_820x400

**Comment by nabaco** [comment]
W_2020-06-07_820x400

Fixed the issue. Required to change how the menus were drawn in a loop, and the logic of the callback since it didn't get saved properly.

In general, this entire thing needs to be refactored and properly documented. Those linked lists gave me a head ache.

**Comment by nabaco** [comment]
Hi @DMJC, my apologies for the mess in the branch. For some reason I could push the changes from the commandline, so I had to do it manually from Github's WebUI.
Yo can squash the commits if you wish. Either way, though changes should fix the issue.
@stephengtuggy @BenjamenMeyer feel free to test it locally.

To compile just this executable and not the entire game run:
`cmake --build . --target vegasettings` in the build dir

**Comment by BenjamenMeyer** [comment]
@nabaco I'll review later today.
Please file an issue for the refactor needs you found

File: engine/FindGTK3.cmake [hunk 4]
```diff
@@ -0,0 +1,611 @@
...
+#   * Run git diff against this file to see all changes
+#=============================================================
+# _GTK3_GET_VERSION
+# Internal function to parse the version number in gtkversion.h
+#   _OUT_major = Major version number
+#   _OUT_minor = Minor version number
+#   _OUT_micro = Micro version number
+#   _gtkversion_hdr = Header file to parse
+#=============================================================
+function(_GTK3_GET_VERSION _OUT_major _OUT_minor _OUT_micro _gtkversion_hdr)
```
Comment (BenjamenMeyer) [inline engine/FindGTK3.cmake:79]: Please make all CMake directives upper case; variables can be lowercase for internal usage or uppercase for constants or exported variables.

File: engine/setup/src/include/central.cpp [hunk 5]
```diff
@@ -89,3 +89,17 @@ struct group *GetGroupStruct(char *name) {
         } while ((CUR = CUR->next) != nullptr);
         return 0;
 }
+
+struct catagory *GetNameFromInfo(char *info) {
+	struct catagory *CUR;
```
Comment (BenjamenMeyer) [inline engine/setup/src/include/central.cpp:94]: looks like tabs were used here; please change to 4 spaces/tab.

File: engine/setup/src/include/display_gtk.cpp [hunk 6]
```diff
@@ -108,15 +106,14 @@ void ShowMain(void) {
...
-//				printf("|||%s|||",GetInfo(NEWCUR->name));
-				gtk_option_menu_set_history(GTK_OPTION_MENU(menu),i);
 				break;
 			}
 			i++;
 		} while ((NEWCUR = NEWCUR->next) != nullptr);
 //		printf("\n\n");
 		gtk_widget_show (menu);
 		gtk_container_add(GTK_CONTAINER(vbox), menu);
+        g_signal_connect(G_OBJECT(menu), "changed", G_CALLBACK(ClickButton), CURRENT);
```
Comment (BenjamenMeyer) [inline engine/setup/src/include/display_gtk.cpp:116]: uniform tab/space

**Comment by nabaco** [comment]
Closing due to author's inactivity. We'll merge the same changes in #165, whilst maintaining original authorship.

**Pull request closed without merging**